Assets/_BuildingGame/Scripts/Building System/BuildState.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace BuildingGame.BuildingSystem
+{
+    public class BuildState : IBuildingState
+    {
+        private List<SnappingPoint> _snappingPoints;
+        private int _currentSelectedSnappingPoint;
+        private int _maxSnappingPoint;
+
+        public void Exit(BuildingBehaviour buildingBehaviour)
+        {
+            buildingBehaviour.CurrentSelectedStructure.DestroyPreviewStructure();
+        }
+
+        public void Start(BuildingBehaviour buildingBehaviour)
+        {
+            buildingBehaviour.CurrentSelectedStructure.SpawnPreviewStructure(PlayerCameraRaycast.RaycastForPoint()); 
+            _currentSelectedSnappingPoint = 0;
+        }
+
+        public void Update(BuildingBehaviour buildingBehaviour)
+        {
+            // Switching the currently selected structure
+            if (buildingBehaviour.inputActions.Building.SwapBuilding.WasPressedThisFrame())
+            {
+                buildingBehaviour.SelectNextStructure();
+                buildingBehaviour.CurrentSelectedStructure.SpawnPreviewStructure(PlayerCameraRaycast.RaycastForPoint());
+            }
+
+            // Getting the target transform and pos to use later
+            Transform hitTransform = PlayerCameraRaycast.RaycastForTransform();
+            Vector3 hitPosition = PlayerCameraRaycast.RaycastForPoint();
+
+            Vector3 targetPos = new Vector3();
+            Vector3 targetRot = new Vector3();
+
+            if(buildingBehaviour.inputActions.Building.Rotate.WasPressedThisFrame()) 
+            { 
+                _currentSelectedSnappingPoint += 1;
+                if (_currentSelectedSnappingPoint > 3) { _currentSelectedSnappingPoint = 0; }
+            }
+
+            if(hitTransform != null)
+            {
+                // Change target depending on you look at a structure or ground
+                if (hitTransform.tag == "Structure")
```

Comment:
```suggestion
                if (hitTransform.tag == STRUCTURE_TAG)
```
It is better to use a const string variable for "Structure", because doing a string comparison (especially in an update loop) can be expensive.